src/outlineView.ts

```diff
@@ -29,23 +46,63 @@ export class OutlineView {
     this.outlineContent.classList.add("outline-content")
   }
 
-  destroy() {
+  public destroy() {
     this.element.remove()
   }
 
-  getElement() {
+  public getElement() {
     return this.element
   }
 
-  getTitle() {
+  public getTitle() {
     return "Outline"
   }
 
-  getIconName() {
+  public getIconName() {
     return "list-unordered"
   }
 
-  setOutline(outlineTree: OutlineTree[], editor: TextEditor, isLarge: boolean) {
+  public setOutline(editor: TextEditor | undefined) {
+    return new Promise<void>((resolve) => {
+      if (this.isRendering) {
+        // return if setOutline is already called and not finished yet.
+        resolve()
+        return
```

Comment:
If secondary calls return here the timeout never gets cleared and reset below.

Reply:
But `this.isRendering` becomes false inside the `setTimeout` callback. 🤔